solutions/usaco-899.mdx

```diff
@@ -0,0 +1,113 @@
+---
+id: usaco-899
+source: USACO Gold 2019 January
+title: Shortcut
+author: Ryan Chou
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_shortcut_gold_jan19.html)
+
+## Explanation
+
+A simple brute-force approach would be to find the shortest path for all of the nodes, and then try to add a shortcut from $1$ to $i$ and find the shortest path again for all $i$.
+
+Using this approach, the answer is $\min(\sum\limits_{i=0}^n \texttt{dist}-\texttt{short}_i)$, where $\texttt{dist}$ is the travel time that all cows take without a shortcut and $\texttt{short}_i$ represents the travel time that all cows take when there is a shortcut from $1$ to $i$. Although this will certainly TLE, it hints at a faster way to compute the time saved. 
+
+Let's call $\texttt{occ}_i$ the number of cows that pass through field $i$. Then the answer would be $\max(\sum\limits_{i=0}^n \texttt{occ}_i \cdot (\texttt{trav}_i - T))$. $\texttt{trav}_i$ being the cost for one cow to get from field $i$ to $1$.
```

Comment:
put the `trav` definition before the "then the answer would be" maybe?

Reply:
also elaborate on why this exists i mean like don't just spit random formulas and say "bro trust me"

Reply:
> don't just spit random formulas and say "bro trust me"

<img width="524" alt="image" src="https://user-images.githubusercontent.com/81596991/151115315-98f26c8c-5162-4f3f-bea3-aef1839a2b2f.png">

😆, yeah ill change it